### ISSUE TITLE
Qt Q_GADGET macro is not supported

### ISSUE THREAD

**Issue by FabriceSalvaire** [doxygen/doxygen#9352]
It seems `Q_GADGET` Qt macro is not supported since it generates `Q_GADGETQString Foo::name` for this example:

```
class Foo
{
    // Q_OBJECT // is discarded / muted in the output
    Q_GADGET // is interpreted as a type of the following code
// public: // is a workaround to break the parser
    /// A name
    Q_PROPERTY(QString name READ name CONSTANT)
}
```

**Comment by albert-github** [comment]
Please always mention the doxygen version used, this problem is present in the current version: 1.9.4 (5d15657a55555e6181a7830a5c723af75e7577e2)

Not all `Q_...` type of macros are supported by doxygen.
It is quite easy for a user to "support" the `Q_GADGET` macro by means of the settings:
```
PREDEFINED += Q_GADGET
MACRO_EXPANSION = YES
```

As the `Q_GADGET` is quite close to the `Q_OBJECT` macro (see: https://doc.qt.io/qt-5/qobject.html#Q_GADGET): 
>The Q_GADGET macro is a lighter version of the [Q_OBJECT](https://doc.qt.io/qt-5/qobject.html#Q_OBJECT) macro for classes that do not inherit from [QObject](https://doc.qt.io/qt-5/qobject.html) but still want to use some of the reflection capabilities offered by [QMetaObject](https://doc.qt.io/qt-5/qmetaobject.html). Just like the [Q_OBJECT](https://doc.qt.io/qt-5/qobject.html#Q_OBJECT) macro, it must appear in the private section of a class definition.

 I think it is good to automatically support it, hence: I've just pushed a proposed patch, pull request #9355

It is up to @doxygen to decide whether the `PREDEFINED` version should be used by the user or to merge the proposed pull request.

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.5.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
